src/Build/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -242,11 +242,24 @@ public NodeEngineShutdownReason Run(bool enableReuse, out Exception shutdownExce
         /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
         /// <returns>The reason for shutting down.</returns>
         public NodeEngineShutdownReason Run(bool enableReuse, bool lowPriority, out Exception shutdownException)
+        {
+            return Run(enableReuse, lowPriority, true, out shutdownException);
+        }
+
+        /// <summary>
+        /// Starts up the node and processes messages until the node is requested to shut down.
+        /// </summary>
+        /// <param name="enableReuse">Whether this node is eligible for reuse later.</param>
+        /// <param name="lowPriority">Whether this node should be running with low priority.</param>
+        /// <param name="workerNode">Whether this node should act as worker node and accpet build requests.</param>
+        /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
+        /// <returns>The reason for shutting down.</returns>
+        public NodeEngineShutdownReason Run(bool enableReuse, bool lowPriority, bool workerNode, out Exception shutdownException)
```

Comment:
this return pattern is not very csharpie.  just wondering why not throw the exception in the out OR include the exception state in the return type?

Reply:
I'd actually prefer to throw the exception, but it's more important to follow what the previous API was, especially since this is part of our public API surface.